bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/account/SmartherNotificationServlet.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.smarther.internal.account;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.smarther.internal.api.dto.Notification;
+import org.openhab.binding.smarther.internal.api.exception.SmartherGatewayException;
+import org.openhab.binding.smarther.internal.util.ModelUtil;
+import org.openhab.binding.smarther.internal.util.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.reflect.TypeToken;
+
+/**
+ * The {@code SmartherNotificationServlet} class acts as the registered endpoint to receive module status notifications
+ * from the Legrand/Bticino C2C Webhook notification service.
+ *
+ * @author Fabio Possieri - Initial contribution
+ */
+@NonNullByDefault
+public class SmartherNotificationServlet extends HttpServlet {
+
+    private static final long serialVersionUID = -2474355132186048438L;
+
+    private static final String CONTENT_TYPE = "application/json;charset=UTF-8";
+    private static final String OK_RESULT_MSG = "{\"result\":0}";
+
+    private final Logger logger = LoggerFactory.getLogger(SmartherNotificationServlet.class);
+
+    private final SmartherAccountService accountService;
+
+    /**
+     * Constructs a {@code SmartherNotificationServlet} associated to the given {@link SmartherAccountService} service.
+     *
+     * @param accountService
+     *            the account service to associate to the servlet
+     */
+    public SmartherNotificationServlet(SmartherAccountService accountService) {
+        this.accountService = accountService;
+    }
+
+    @Override
+    protected void doPost(@Nullable HttpServletRequest request, @Nullable HttpServletResponse response)
+            throws ServletException, IOException {
+        if (request != null && response != null) {
+            logger.debug("Notification callback servlet received POST request {}", request.getRequestURI());
+
+            // Handle the received data
+            final String requestBody = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
```

Comment:
Using `BufferedReader.lines()` does not result in the closing of the reader after the operation is complete, the Reader would still need to be closed manually.
Although less concise than what you've written, I feel that the following code would be more correct and more performant.
```suggestion
            final String requestBody; 
            try(Reader reader = request.getReader()){
                StringWriter writer = new StringWriter();
                int c;
                while((c = reader.read()) != -1){
                    writer.write(c);
                }
                requestBody = writer.toString();
            }
```
WDYT?

Reply:
Hmmm... found something similar on StackOverflow: https://stackoverflow.com/questions/1808248/is-is-necessary-to-close-the-input-stream-returned-from-httpservletrequest
Just reporting here the key paragraph, which I tend to agree to:
> The thumb rule in I/O is, if you did not open/create the inputstream source yourself, then you do not necessarily need to close it as well.
> Here you are just wrapping the request's inputstream, so you don't necessarily need to close it.
The Servlet container should be in charge of closing the request stream.

Reply:
hmm, in that case you can just do it without the try-with-resources.

Reply:
I'd tend to accept your suggestion as more performant.
However, it seems to me your code reads `int` tokens from request body instead of `String`s.
Am I missing something ?

Reply:
> It seems to me your code reads int tokens from request body instead of Strings. Am I missing something ?

Each int token is cast to a `char` as part of Writer.write(int).

> I'd tend to accept your approach as more performant, however I'd slightly change it to something like:
In general you would be correct, but the reason that I didn't use an intermediate buffer in my suggestion was because `getReader` returns a `BufferedReader` already, which means that data will be read by bulk for you behind the scenes. Because the reader is already doing buffering, there is no need for us to do buffering of our own as well.

Reply:
I agree. Changed.